load.php

```diff
@@ -168,6 +186,38 @@ function perflab_render_generator() {
 }
 add_action( 'wp_head', 'perflab_render_generator' );
 
+/**
+ * Don't load the performance modules if the core version of the module is available.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $module The name of the module.
+ * @return bool Whether to load module or not.
+ */
+function perflab_can_load_module( $module ) {
+	$module_load_file = plugin_dir_path( __FILE__ ) . 'modules/' . $module . '/can-load.php';
+
+	// If the load file does not exist, the feature is not in core and the module can be loaded.
+	if ( ! file_exists( $module_load_file ) ) {
+		return true;
+	}
+
+	// Require the file to include the add_filter function.
+	require_once $module_load_file;
+
+	/**
+	 * Filters whether the module can load or not.
+	 *
+	 * You can set this to false in order to disable the module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param bool   $can_load_module Whether to load module. default true.
+	 * @param string $module          The name of the module.
+	 */
+	return apply_filters( 'perflab_can_load_module', true, $module );
```

Comment:
This implementation is different from the proposed one in the issue. Could you explain why you went with that approach instead?

I think using a filter here makes this more complicated than it needs to be. To clarify, what was outlined in the issue would be the following:
```suggestion
	// Require the file to get the closure for whether the module can load.
	$can_load = require $module_load_file;

	// If the `can-load.php` file is invalid and does not return a closure, assume the module can be loaded.
	if ( ! is_callable( $can_load ) ) {
		return true;
	}

	// Call the closure to determine whether the module can be loaded.
	return $can_load();
```

Reply:
At first, I implement it with the function based solution [here](https://github.com/mukeshpanchal27/performance/pull/2/commits/cfd8b5f27732376f0c3edfbe06c9ad48e9d0bcec) but in some scenarios, it was given me the wrong return value for the module function when it calls so to scale the functionality I implement filter based function.

I agreed that this code snippet makes more sense than the filter and it has less complex. I'm going to update it and mark this conversation as resolved. Thanks!